src/lib/shared/payloads.ts

```diff
@@ -0,0 +1,57 @@
+import type { TransactionResponseJSON } from '@hashgraph/sdk'
+
+/*
+ * Request params
+ *
+ */
+
+export type SignTransactionAndSendParams = {
+  signerAccountId: string
+  signedTransaction: string
+}
+
+export type SendTransactionOnlyParams = {
+  signedTransaction: string
+}
+
+export type SignTransactionBodyParams = {
+  signerAccountId: string
+  transactionBody: string
+}
+
+export type GetNodeAddressesParams = undefined
+
+export type SignMessageParams = {
+  message: string
+}
+
+export type QueryParams = {
+  query: string
+}
+
+/*
+ * Responses
+ * {
+ *  "precheckCode": "<an integer representing the [ResponseCodeEnum] value returned from the Hedera Node, which may indicate success or failure.>",
+ *  "transactionId": "<a string encoded transaction identifier of the signed message that was sent to the Hedera node, in <shard>.<realm>.<number>@<seconds>.<nanos> format.>",
+ *  "nodeId": "<a string encoded transaction identifier of the Hedera Gossip Node’s Account that the transaction was submitted to, in <shard>.<realm>.<number> format>",
+ *  "transactionHash": "<base64 encoding of the SHA384 digest of the signedTransactionBytes of the Transaction message that was submitted to the Hedera Network.>"
+ * }
+ */
+
+export type SignTransactionAndSendResponse = TransactionResponseJSON & { precheckCode: number }
```

Comment:
Precheck code on a successful transaction seems pointless? Should only be returned with error, also not sure how to get precheck code from a transaction response

Reply:
@bugbytesinc

Reply:
Also commented on HIP-820 discussion, in essence, we are operating at a higher level than raw JSON-RPC with session requests.  Since the wallet will have succeeded in sending the TX to the network by having received a precheck code (and not a GRPC/Https exception) then returning the code (regardless of its value) seems proper.  I realize the JS SDK due to its legacy and complexity may be in a tug-of-war with the HAPI, but if the SDK throws a PrecheckStatusError, it can be caught and contains the necessary values to return over the wire, including the precheck code.   The dAPP can then interpret non zero values as rejections, but it will know very specifically why.